homeassistant/components/yolink/__init__.py

```diff
@@ -95,7 +109,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             hass.data[DOMAIN][entry.entry_id][ATTR_DEVICE] = yolink_devices.data[
                 ATTR_DEVICE
             ]
-            await yolink_mqtt_client.init_home_connection(home_info.data["id"])
+            await yolink_mqtt_client.init_home_connection(
+                home_info.data["id"], coordinator.async_set_updated_data
```

Comment:
Please don't pass the coordinator callback directly to the library client. Make an intermediate function here in the integration that we can pass to the library client. The intermediate function should call the coordinator callback.

This ensures that this integration retains control of the interface between the integration and the library.